src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -128,6 +124,18 @@ class AgoraBusiness {
     }
   }
 
+  private def validateMethodRef(method: AgoraEntity, username: String) = {
+    if (!AgoraEntityPermissionsClient.getEntityPermission(method, username).canRead &&
+      !AgoraEntityPermissionsClient.getEntityPermission(method, "public").canRead) {
+      throw new AgoraEntityNotFoundException(method)
```

Comment:
We are intentionally being ambiguous here because we don't want the user to be able to tell if a method truly doesn't exist vs exists but not readable to them. So we return a 404 in either case.